openedx/openedx-platform
# Only send course update if student completes a block

This change ensures that the course highlights are only sent to a student if they have completed atleast one block of the course.

**JIRA Ticket:** [OSPR-4213](https://openedx.atlassian.net/browse/OSPR-4213)

**Sandbox URL:**
LMS: https://pr23359.sandbox.opencraft.hosting/
Studio: https://studio.pr23359.sandbox.opencraft.hosting/

Note: We can help with testing by modifying user accounts if needed.

**Testing Instructions:**
1. Create a self-paced course and enable highlights. The sandbox already contains a sample course with highlights enabled.
2. Ensure that the completion tracking waffle switch is enabled.
3. As a student, enroll into the course but don't start it.
4. Modify the start date on the schedules object to a few weeks prior.
5. Run the management command to send course update with `check-completion` flag set to `True` and verify that student does not receive the update.
6. Start the course and complete a block.
7. Run the management command to send course update again and verify that the student recevies the first week update.


**Reviewers:**
